src/.clang-tidy

```diff
@@ -0,0 +1,83 @@
+# Some checks are highly style dependant. The goal is NOT to activate all
+# of them.
+
+---
+Checks:          '-*,
+                  readability-*,
+                  -readability-implicit-bool-conversion,
+                  -readability-magic-numbers,
+                  -readability-else-after-return,
+                  -readability-named-parameter,
+                  -readability-qualified-auto,
+                  -readability-magic-numbers,
+                  -readability-isolate-declaration,
+                  -readability-inconsistent-declaration-parameter-name,
+                  -readability-simplify-boolean-expr,
+                  performance-*,
+                  -performance-unnecessary-value-param,
+                  -performance-faster-string-find,
+                  modernize-*,
+                  -modernize-avoid-c-arrays,
+                  -modernize-deprecated-headers,
+                  -modernize-loop-convert,
+                  -modernize-pass-by-value,
+                  -modernize-use-auto,
+                  -modernize-use-bool-literals,
+                  -modernize-use-using,
+                  -modernize-use-trailing-return-type,
+                  cppcoreguidelines-*,
+                  -cppcoreguidelines-pro-bounds-array-to-pointer-decay,
+                  -cppcoreguidelines-pro-type-vararg,
+                  -cppcoreguidelines-owning-memory,
+                  -cppcoreguidelines-avoid-magic-numbers,
+                  -cppcoreguidelines-pro-bounds-pointer-arithmetic,
+                  -cppcoreguidelines-no-malloc,
+                  -cppcoreguidelines-init-variables,
+                  -cppcoreguidelines-avoid-c-arrays,
+                  -cppcoreguidelines-pro-bounds-constant-array-index,
+                  -cppcoreguidelines-pro-type-cstyle-cast,
+                  -cppcoreguidelines-avoid-goto,
+                  -cppcoreguidelines-pro-type-member-init,
+                  -cppcoreguidelines-macro-usage,
+                  -cppcoreguidelines-pro-type-reinterpret-cast,
+                  -cppcoreguidelines-pro-type-union-access,
+                  -cppcoreguidelines-narrowing-conversions,
+                  bugprone-*,
+                  -bugprone-signed-char-misuse,
+                  -bugprone-branch-clone,
+                  -bugprone-narrowing-conversions,
+                  cert-*,
+                  -cert-err34-c,
+                  -cert-dcl50-cpp,
+                  -cert-err58-cpp,
+                  clang-diagnostic-*,
+                  clang-analyzer-*,
+                  -clang-analyzer-alpha*,
+                  -clang-analyzer-valist.Uninitialized,
+                  -clang-analyzer-optin.performance.Padding'
+WarningsAsErrors: '*'
+# Exclude subdirectories (currently there is only thrid_party)
+HeaderFilterRegex: 'src/[^\\/]*^'
+CheckOptions:    
+  - key:             readability-braces-around-statements.ShortStatementLines
+    value:           2
```

Comment:
Just curious, why is this needed? If I interpret the documentation correctly the default value (0) should be enough (and what I would like).

Reply:
If you prefer 0 than the default is fine :-)
I had done a fix there in parallel and was considering to change the default to 2. So I just made the default explicit here.

Reply:
Fixed.